tests/test_utils.rs

```diff
@@ -0,0 +1,7 @@
+pub async fn determine_port_to_bind() -> async_std::net::SocketAddr {
```

Comment:
Perhaps a shorter name would be `find_port`.

Reply:
as with above comment, would `find_socket_addr` be reasonably concise and clear? it's really something like "find a port and return a socket addr with that port" but that's awkward af